jep/0000/README.adoc

```diff
@@ -0,0 +1,232 @@
+
+== Specification
+
+=== Global overview
+
+The setup uses an link:https://aws.amazon.com/cloudformation[AWS _CloudFormation_ template] that instantiates the right resources.
+
+It creates an EC2 instance, to host the Jenkins Essentials master in a dedicated Instance Profile.
+The master will be able to provision agents from EC2, and store artifacts directly in S3.
+
+=== Used plugins and configuration
+
+* link:https://plugins.jenkins.io/ec2[EC2 plugin]: to provide dynamically provisioned agents
+  footnote:[even more important for Jenkins Essentials which is configured with the link:JENKINS-49861[sane 0-executor for the master node]].
+* link:https://plugins.jenkins.io/artifact-manager-s3[Artifact Manager on S3 plugin]: to store artifacts and Pipeline stashes in Amazon S3.
+
+// JENKINS-52342 for ^ which is the design, but not done fully yet.
+
+As a general rule for Essentials, we are using the `configuration-as-code` plugin to do the necessary configuration.
+
+*No credentials* are specified.
+The permissions are retrieved dynamically through the instance profile used when creating the EC2 instance for the Essentials master.
+
+=== Instance Profile and IAM role
+
+A dedicated instance profile is created by the _CloudFormation_ template to allow creating EC2 instances for agents, and access a specific S3 bucket to store artifacts and Pipeline stashes.
+
+==== EC2 policy
+
+[source,json]
+"Statement": [{
+    "Sid": "ControlEC2JenkinsAgents",
+    "Effect": "Allow",
+    "Action": [
+        "ec2:DescribeInstances",
+        "ec2:TerminateInstances",
+        "ec2:RequestSpotInstances",
+        "ec2:DeleteTags",
+        "ec2:CreateTags",
+        "ec2:DescribeRegions",
+        "ec2:RunInstances",
+        "ec2:DescribeSpotInstanceRequests",
+        "ec2:StopInstances",
+        "ec2:DescribeSecurityGroups",
+        "ec2:GetConsoleOutput",
+        "ec2:DescribeImages",
+        "ec2:CancelSpotInstanceRequests",
+        "ec2:StartInstances",
+        "ec2:DescribeAvailabilityZones",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeKeyPairs"
+    ],
+    "Resource": "*"
+}]
+
+==== S3 bucket policy
+
+FIXME: refine through JENKINS-52342.
+
+[source,json]
+"Statement": [{
+        "Sid": "TodoRefineSecurityALot",
+        "Effect": "Allow",
+        "Action": [
+            "s3:PutObject",
+            "s3:GetObject",
+            "s3:ListBucket",
+            "s3:DeleteObject"
+        ],
+        "Resource": "*"
+    }
+]
+
+=== Security groups
+
+==== Master
+A dedicated group is created to allow access to the following ports:
+
+22:: restricted to an optional parameter.
+Users would typically use `curl ident.me` to avoid making this port accessible to the world.
+8080:: not restricted.
+50000:: not restricted (for agents).
```

Comment:
Yes. It would be an improvement. Though I'm finally not even sure we need that port, see below in the secu part. Need to check that.

Reply:
OK, I just checked and this port is actually useless. I'm not that surprised as IIUC this goes through the same pipe as when using manually the [SSH Agent plugin](https://wiki.jenkins.io/display/JENKINS/SSH+Agent+Plugin), but I had never checked. Thanks for challenging it @dcoraboeuf. Going to fix this. Will make it more secure, and simpler in one go \o/.

Reply:
@dcoraboeuf, oh, and thank you again. In case you or anyone wonder: your input was **very** valuable. Even deciding to close as WONTFIX is already an important improvement, instead of just doing things for no clear reason, your challenge pushed us to actually think more deeply about if and why we did want to keep that port open.